cmake/Modules/FindFILESYSTEM.cmake

```diff
@@ -0,0 +1,240 @@
+#[=======================================================================[.rst:
```

Comment:
Did you copy this file from somewhere? In this case, do we need a special license header for this file?

Reply:
I took the file from https://github.com/vector-of-bool/CMakeCM/blob/master/modules/FindFilesystem.cmake , and then I did some modifications. The original license is BSD 3-Clause.  Should I include that in the header?

Reply:
Yeah. I think you should do so. My takeaway from other license header discussion is that we should put the GPL header first because some changes were made and in this case followed by `BSD 3-Clause`.

Reply:
agreed with @jdemel !

Reply:
Please let me know if the header is now _how it should be_. In my understanding, it is ok to release under the GPL a derived work from a BSD-3 code, as long as you retain the original copyright notice. But I really would like to confirm with you guys that this is the right way of doing it, or if there is a more adequate wording for this case.

Reply:
I'd recommend including `See https://cmake.org/licensing for details.` as should have been in the original.

Reply:
The GPL header looks good to me; thanks!

Reply:
Included `See https://cmake.org/licensing for details.` in f7e2d9bcbf84fd4151b8eb088c13f449abcd84d3

Reply:
@bhilburn In your understanding of licensing, is this header info (1) sufficient both for the GPL; (2) in keeping enough of the prior license as need be; (3) in allow Carles to note that he changed the original file? Thx!

Reply:
I'll give Ben through the end of this work-week to respond. If not, we'll go with this license as current & fix it up if/as need be later.

Reply:
Yup, current version is good 👍

Reply:
great; thanks @bhilburn !